app/src/main/java/com/owncloud/android/ui/adapter/OCFileListAdapter.java

```diff
@@ -740,6 +740,14 @@ public void setData(List<Object> objects,
         new Handler(Looper.getMainLooper()).post(this::notifyDataSetChanged);
     }
 
+    @SuppressLint("NotifyDataSetChanged")
+    public void replaceFile(OCFile file) {
+        mFiles.remove(file);
+        mFiles.add(file);
+        mFiles = sortOrder.sortCloudFiles(mFiles);
+        notifyDataSetChanged();
+    }
```

Comment:
```suggestion
    public void replaceFile(OCFile file) {
        mFiles.set(mFiles.indexOf(file), file);
        notifyItemChanged(file);
    }
```

If I'm not mistaken, the sorting order should remain the same. So it should suffice to replace and update only our selected file/item.